### PR TITLE
Pass through subscription options to google cloud client

### PR DESCRIPTION
**Pull request by ghost**
We've run into a case where we want to set subscription options (specifically `messageRetentionDuration` and `expirationPolicy.ttl`) and the subscription options were not being passed all the way down to the google cloud pub sub client.

Example use case:

```typescript
const subOptions = {
    	messageRetentionDuration: { seconds: 86400 },
	expirationPolicy: {
		ttl: { seconds: 86400 },
	},
};

const client = new GooglePubSub(...);
await client.asyncIterator("abc123", subOptions);
```

I modified the pub sub 'engine' to pass those options through to the point where this library interacts with that client.

Happy to update and docs/tests or make other changes. This just seemed like a quick and easy change :)

### CODE REVIEW

**Comment by jonas-app** [comment]
Hey @jacobcabantomski-ct, thx for your contribution. 👏 
I noticed the same issue a couple month ago but never invested the time fixing it as we haven't used these options till know. 😇 
Happy to merge this and release a new version.
Would be perfect if you could adjust the readme & add a test. 💯

**Review by jonas-app — APPROVED** [review]
@jacobcabantomski-ct 👍 
Can you update the commit message?

- Fix the typo (_subscriptuion_)
- Use a short header and a detailed description (https://chris.beams.io/posts/git-commit/)

And a test would be amazing.
Thx again for your contribution.

File: README.md [hunk 1]
```diff
@@ -119,7 +119,7 @@ const pubSub = new GooglePubSub(options, topic2SubName, commonMessageHandler)
 ```
 
 ### Options
-This are the [options](https://cloud.google.com/nodejs/docs/reference/pubsub/0.23.x/global#ClientConfig) which are passed to the internal Google PubSub client.
+This are the [options](https://googleapis.dev/nodejs/pubsub/1.1.5/global.html#ClientConfig) which are passed to the internal Google PubSub client.
```
Comment (jonas-app) [inline README.md]:
```suggestion
These are the [options](https://googleapis.dev/nodejs/pubsub/1.1.5/global.html#ClientConfig) which are passed to the internal or passed Google PubSub client.
```

**Comment by ghost** [comment]
@jonas-arkulpa Still adding tests, will cleanup per your recommendations. Thanks!

**Comment by ghost** [comment]
@jonas-arkulpa Comments addressed, tests added. Ready for your re-review.

**Comment by jonas-app** [comment]
@jacobcabantomski-ct Merged and released a new version [2.1.0](https://www.npmjs.com/package/@axelspringer/graphql-google-pubsub/v/2.1.0) to npm. Thanks!

**Pull request merged**